standalone/src/main/java/io/delta/standalone/actions/Metadata.java

```
@@ -133,6 +139,58 @@ public StructType getSchema() {
+                .map(entry -> new Constraint(
+                                entry.getKey().replace(Constraint.CHECK_CONSTRAINT_KEY_PREFIX, ""),
+                                entry.getValue()))
+                .collect(Collectors.toList());
+
+        return checkConstraints;
+    }
+
+    /**
+     * Add the specified check constraint to the {@code configuration}
+     * @param name  the name of the check constraint
+     * @param expression  the condition to enforce as a SQL string
+     * @return a new {@link Metadata} instance with the same properties as this instance but with
+     *         the added check constraint
+     * @throws IllegalArgumentException if a constraint with name already exists
+     */
+    public Metadata addCheckConstraint(String name, String expression) throws Throwable {
+        String fullKey = Constraint.CHECK_CONSTRAINT_KEY_PREFIX + name.toLowerCase(Locale.ROOT);
+        if (configuration.containsKey(fullKey)) {
+            throw DeltaErrors.checkConstraintAlreadyExists(name, configuration.get(fullKey));
+        }
+        Map<String, String> newConfiguration = new HashMap(configuration);
+        newConfiguration.put(fullKey, expression);
+        return copyBuilder().configuration(newConfiguration).build();
+    }
+
+    /**
+     * Removes the specified check constraint from the {@code configuration}
+     * @param name  the name of the check constraint to remove
+     * @return a new {@link Metadata} instance with the same properties as this instance but with
+     *         the specified check constraint removed
+     * @throws IllegalArgumentException if a constraint with name does not exist
+     */
+    public Metadata removeCheckConstraint(String name) throws Throwable {
+        String fullKey = Constraint.CHECK_CONSTRAINT_KEY_PREFIX + name.toLowerCase(Locale.ROOT);
+        if (!configuration.containsKey(fullKey)) {
+            throw DeltaErrors.checkConstraintDoesNotExist(name);
+        }
```

I don't think we need to throw an error here. All the user wants is to ensure that the constraint no longer exists. If it was never there to begin with, that should be fine.

Happy to be convinced otherwise.

---

I just followed what we do in sql when doing with a constraint that DNE
```
alter table table_name drop constraint constraint_name
```

---

IMO the decision to or not to throw an error is at a much higher scope than `Metadata.java`. So I think we should not throw an error here. IMO the connector can do that themselves.

Perhaps we can add a `hasCheckConstraint: Boolean` API so they can efficiently look it up?

WDYT @venki?